src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int capacity;
+    private Node<K, V>[] hashTable;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        hashTable = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > capacity * LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = hash(key);
+        if (hashTable[index] == null) {
+            hashTable[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> current = hashTable[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                current.next = newNode;
+                size++;
```

Comment:
let's add return here as well
it seems like you are adding a node and then go to that added node in a loop and replace its value (this is not a mistake, but it is redundant)